### ISSUE TITLE
API key for gated models

### ISSUE THREAD

**Issue by Omay238** [oobabooga/text-generation-webui#4326]
**Description**

Using download-model.py for Llama 2 doesn't work because it is a gated model. Perhaps a command-line flag or input function.

**Additional Context**
```
Traceback (most recent call last):
  File "/Users/----/Documents/text-generation-webui/download-model.py", line 265, in <module>
    links, sha256, is_lora, is_llamacpp = downloader.get_download_links_from_huggingface(model, branch, text_only=args.text_only, specific_file=specific_file)
                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/----/Documents/text-generation-webui/download-model.py", line 75, in get_download_links_from_huggingface
    r.raise_for_status()
  File "/Users/----/anaconda3/lib/python3.11/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://huggingface.co/api/models/meta-llama/Llama-2-13b-chat-hf/tree/main
```

**Comment by missionfloyd** [comment]
You can set your huggingface username/password with the `HF_USER` and `HF_PASS` environment variables, or API key with `HF_TOKEN`.

Add to the top of `start_windows.bat`:
```batch
set HF_USER=[username]
set HF_PASS=[password]
```
or
```batch
set HF_TOKEN=[API key]
```

**Comment by Tum1370** [comment]
I tried this and it does not work. I Have added 
et HF_USER=[username]
set HF_PASS=[password]
to the top of my start_windows batch (just above the @echo off command at top of this file.
I have got permission on my HUggin Face account for the model i am trying to download. Yet still all i see is errors while its downloading. Then the model does not load.

The model am trying to download is the following.

https://huggingface.co/meta-llama/Llama-3.2-1B

I am not sure what else i can do.
